test/unit/org/apache/cassandra/utils/CassandraGenerators.java

```diff
@@ -650,13 +1024,32 @@ public static Gen<IPartitioner> partitioners()
                         .flatMap(SupportedPartitioners::partitioner);
     }
 
+
     public static Gen<IPartitioner> nonLocalPartitioners()
     {
         return SourceDSL.arbitrary().enumValues(SupportedPartitioners.class)
                         .assuming(p -> p != SupportedPartitioners.Local)
                         .flatMap(SupportedPartitioners::partitioner);
     }
 
+    /**
+     * For {@link LocalPartitioner} it can have a very complex type which can lead to generating data larger than
+     * allowed in a primary key.  If a test needs to filter out those cases, can just
+     * {@code .map(CassandraGenerators::simplify)} to resolve.
+     */
+    public static IPartitioner simplify(IPartitioner partitioner)
```

Comment:
in a follow up patch ill prob remove this... `LocalPartitioner` has caused flakyness for several tests in cep-15-accord so plan to refactor that logic to be less problematic, which lets us drop this function... keeping for now just to be in-sync